paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/operator/ReadOperator.java

```diff
@@ -130,6 +137,14 @@ public void processElement(StreamRecord<Split> record) throws Exception {
                     reuseRecord.replace(nestedProjectedRowData);
                 }
                 output.collect(reuseRecord);
+
+                if (limit != null && numRecordsIn.getCount() == limit) {
+                    LOG.info(
+                            "When limit is not null, reader {} reach the limit record {}.",
+                            this,
+                            limit);
+                    return;
```

Comment:
When will this condition be met? limit should have been applied to the ReadBuilder in FlinkSourceBuilder#createReadBuilder, so the limit should not come into effect here.

Reply:
@yunfengzhou-hub 
Hi, thanks for your reply.

The reason why I add this condition here is :
The reader should end as soon as possible after reading the limit data, it should not read unnecessary data.